src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -109,8 +109,25 @@ struct OperationParameters {
         }
 
         var signatureParameterViewModel = [ParameterViewModel]()
-        for param in parameters.inSignature {
-            signatureParameterViewModel.append(ParameterViewModel(from: param))
+        for param in parameters {
+            // For `Options` Group schema created by "x-ms-parameter-grouping" with postfix: Options,
+            // look for  all the properties from that group schema which is in 'path' as the signature of the method
+            if let group = model.schemas.schema(for: param.name, withType: .group) as? GroupSchema,
+                group.name.hasSuffix("Options") {
+                for property in group.properties ?? [] {
+                    switch property {
+                    case let .grouped(groupProperty):
+                        for param in groupProperty.originalParameter where param.paramLocation == .path {
+                            signatureParameterViewModel
+                                .append(ParameterViewModel(from: param))
+                        }
+                    default:
+                        continue
+                    }
+                }
+            } else if param.belongsInSignature() {
+                signatureParameterViewModel.append(ParameterViewModel(from: param))
+            }
```

Comment:
I wonder if this check should be done when the model is created. It seems strange to need to pass the model in here. If these params that should be in the options model were processed and popped earlier, we wouldn't need the reference to the model here.

Reply:
This check is needed for building signature of the operation, not for creating the model file.

This is needed as there are scenario where

Like in Paging, it has this `GroupProperty`:
Technically, it should be folded into method, Option object. However, the parameter `offset` is in `path`, so that should not be part of the Option object. So we should put that in signature parameters , like all other `path` parameters.
```
          originalParameter:
            - !<!Parameter> &ref_44
              schema: *ref_2
              groupedBy: *ref_9
              implementation: Method
              required: true
              language: !<!Languages> 
                default:
                  name: offset
                  description: Offset of return value
                  serializedName: offset
              protocol: !<!Protocols> 
                http: !<!HttpParameter> 
                  in: path
```